server/src/test/java/org/elasticsearch/rest/RestControllerTests.java

```diff
@@ -640,6 +641,41 @@ public boolean compatibilityRequired() {
         assertTrue(channel.getSendResponseCalled());
     }
 
+    public void testIncorrectCompatibleHandlersDoNotDispatch() {
+        final byte version = (byte) (Version.CURRENT.major - 1);
+
+        final String mimeType = randomFrom("application/vnd.elasticsearch+json;compatible-with="+version);
+        String content = randomAlphaOfLength((int) Math.round(BREAKER_LIMIT.getBytes() / inFlightRequestsBreaker.getOverhead()));
+        final List<String> contentTypeHeader = Collections.singletonList(mimeType);
+
+        // request to compatible api with body requires a content-type header.
+        // this one does no have it, making it non compatiblegit
+        FakeRestRequest fakeRestRequest = new FakeRestRequest.Builder(NamedXContentRegistry.EMPTY)
+            .withContent(new BytesArray(content), RestRequest.parseContentType(contentTypeHeader)).withPath("/foo")
+            .withHeaders(Map.of("Accept", contentTypeHeader))
+            .build();
```

Comment:
doesn't this request have content-type and accept header and a body ? (the body is withContent right?)

So with a body, matching accept and content-type headers , shouldn't this be an example of a good request ?

Reply:
good point, contentTypeHeader misleadingly named,
withContent` also sets `Content-Type` behind the scenes, but it is the overriden by a withHeaders method that only has Accept header
will rename this to
```
            .withHeaders(Map.of("Accept", mimeTypeList))
```

Reply:
thanks, I missed that.